docs/architecture/adr-038-add-support-for-wasm-light-clients.md

```diff
@@ -0,0 +1,86 @@
+# ADR 38: Add support for Wasm based light client
+
+## Changelog
+
+- 26/11/2020: Initial Draft
+
+## Status
+
+Draft
+
+## Abstract
+
+Currently in the SDK, light clients are part of Cosmos SDK codebase. What this means that, anytime a blockchain built 
+upon Cosmos SDK need to add support for new light client, Cosmos SDK need to be modified and all validator nodes of that 
+blockchain need to be updated with custom version of Cosmos SDK. 
```

Comment:
I do not quite understand this? What do you mean by they're part of the SDK codebase? And how does this relate to specific apps?

The light client is implemented by Tendermint and used by the SDK in some capacity (e.g. as an external process).

Reply:
Ohh, this is IBC specific? That needs to be clearly stipulated in both the context/background and title.

Reply:
yea, I prefer the terms `on-chain light client` to refer to IBC light clients and `off-chain light client` to refer to the Tendermint one

Reply:
or just `IBC light client` or `IBC client`